lib/rules/default-param-last.js

```diff
@@ -5,6 +5,35 @@
 
 "use strict";
 
+/**
+ * checks if node is optional parameter
+ * @param {ASTNode} node the node to be evaluated
+ * @returns {boolean} whether the node is optional
+ */
+function isOptionalParam(node) {
+    return (
+        (node.type === "ArrayPattern" ||
+            node.type === "AssignmentPattern" ||
+            node.type === "Identifier" ||
+            node.type === "ObjectPattern" ||
+            node.type === "RestElement") &&
+        node.optional
+    );
+}
+
+/**
+ * checks if node is plain parameter
```

Comment:
For clarity: Can you explain what "plain" means here? I'm guessing you mean it's just an identifier but I'm not 100% sure.

Reply:
Existing code that I also found confusing. "Plain" meant non-optional, or in other words, required. I switched the naming around.